oioioi/problems/models.py

```diff
@@ -885,6 +886,62 @@ class Meta(object):
         verbose_name_plural = _("difficulty proposals")
 
 
+
+@receiver(post_save, sender=DifficultyTagProposal)
+def increase_aggregated_difficulty_tag_proposal(sender, instance, created, **kwargs):
+    problem = instance.problem
+    tag = instance.tag
+    
+    if created:
+        aggregated_difficulty_tag_proposal = AggregatedDifficultyTagProposal.objects.get_or_create(
+            problem=problem, tag=tag
+        )[0]
+        
+        aggregated_difficulty_tag_proposal.amount += 1
+        aggregated_difficulty_tag_proposal.save()
+
+
+@receiver(post_delete, sender=DifficultyTagProposal)
+def decrease_aggregated_difficulty_tag_proposal(sender, instance, **kwargs):
+    problem = instance.problem
+    tag = instance.tag
+    
+    try:
+        aggregated_difficulty_tag_proposal = AggregatedDifficultyTagProposal.objects.get(problem=problem, tag=tag)
+        aggregated_difficulty_tag_proposal.amount -= 1
+        
+        if aggregated_difficulty_tag_proposal.amount == 0:
+            aggregated_difficulty_tag_proposal.delete()
+        else:
+            try:
+                aggregated_difficulty_tag_proposal.save()
+            except ValidationError as e:
+                raise RuntimeError(
```

Comment:
I suggest to log an error instead of raising an exception

Reply:
Completely rewrote these functions, any caught exceptions in decreasing aggregated tags are logged.